python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -753,6 +763,23 @@ def build_asset_ins(
     return ins_by_asset_key
 
 
+def build_subsettable_asset_ins(
+    asset_ins: Mapping[AssetKey, Tuple[str, In]],
+    asset_outs: Mapping[AssetKey, Tuple[str, Out]],
+    upstream_deps: Iterable[AbstractSet[AssetKey]],
+) -> Mapping[AssetKey, Tuple[str, In]]:
+    """Creates a mapping from AssetKey to (name of input, In object) for any asset key that is not
+    currently an input, but may become one if this asset is subset.
```

Comment:
A tiny example would be helpful here for explaining this.  E.g. if a subsettable multi-asset includes A->B, and we create a subset that only includes B, we'd want an input corresponding to A. Because there are some situations where A would be generated by a different step within the same job.